### PR TITLE
Reintroduce validation to access scopes within roxctl

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

This is a follow-up from the previous PR which disabled the validation within the `transform` implementation for access scopes.

The validation was removed due to no integration health being created for the access scope in case of the reconciler failing. This would lead to the resource being not discoverable within the integration health status.

Now, the validation has been split from the transforming part and the code for validating an access scope has been shared between the datastore and the roxctl command.

Going forward, we should think how we treat these kind of validations: Shall we move everything to a shared location, so that all consumers (roxctl, sensor, central datastores) can re-use their validation? Should the database schema cover this?

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

- unit tests added.
- the E2E tests for declarative config should still pass.

### CODE REVIEW

**Comment by dhaus67** [comment]
Current dependencies on/for this PR:
* master
  * **PR #5602** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/5602" target="_blank">Graphite</a> 
    * **PR #5604** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/5604" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/5604?utm_source=stack-comment).

**Review by janisz — APPROVED** [review]

File: central/role/validate.go [hunk 1]
```diff
@@ -152,50 +151,10 @@ func ValidateSimpleAccessScope(scope *storage.SimpleAccessScope) error {
 		multiErr = multierror.Append(multiErr, errors.New("name field must be set"))
 	}
 
-	err := ValidateSimpleAccessScopeRules(scope.GetRules())
+	err := role.ValidateSimpleAccessScopeRules(scope.GetRules())
 	if err != nil {
 		multiErr = multierror.Append(multiErr, err)
 	}
```
Comment (ivan-degtiarenko) [inline central/role/validate.go:157]: nit: I think we can rewrite `ValidateSimpleAccessScope` a little bit into `ValidateSimpleAccessScope`(has access scope id check and calls next function) and `ValidateSimpleAccessScopeProto`(checks name + rules).

That way we separate storage constraints(id) from proto struct constraints(name, rules)

File: pkg/role/validate.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package role
```
Comment (ivan-degtiarenko) [inline pkg/role/validate.go:1]: Not sure if package should be `role` and not `accessscope` here

**Review by ivan-degtiarenko — APPROVED** [review]
LGTM modulo minor comments

File: central/role/validate.go (hunk 1 above)
Comment (dhaus67) [reply central/role/validate.go:157]: Made the change, but had to also leave two public methods: one for validating the `rules` proto, as this was required by the `role/service`, and the other validating the whole `simple access scope proto`.

**Pull request merged**